admin/src/scenes/dashboardV2/components/graphs/SessionHorizontalBar.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom"; // Assuming you are using react-router
+import { getGraphColors } from "./graph-commons";
+import GraphTooltip from "./GraphTooltip";
+
+interface SessionHorizontalBarProps {
+  title: string;
+  values: number[];
+  labels: string[];
+  showTooltips: boolean;
+  legendUrls?: string[];
+  goal: number;
+  className?: string;
+}
+
+interface BarProps {
+  color: string;
+  label: string;
+  value: number;
+  percent: string;
+  width: number;
+  tooltip: string | null;
+}
+
+export default function SessionHorizontalBar({ title, values, labels, showTooltips = false, goal, className = "" }: SessionHorizontalBarProps) {
+  const [bars, setBars] = useState<BarProps[]>([]);
+  const [total, setTotal] = useState<number>(0);
+  const [x100, setX100] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (values && values.length > 0) {
+      const total = values.reduce((value, originalValue) => value + originalValue, 0);
+      setTotal(total);
+      const localGoal = goal === 0 ? total : goal;
+      const colors = ["#1D4ED8", "#E5E7EB"];
+
+      setBars(
+        values.map((value, idx) => ({
+          color: colors[idx % colors.length],
+          label: labels[idx],
+          value,
+          percent: localGoal === 0 ? "-" : `${Math.round((value / localGoal) * 100)}%`,
+          width: Math.min(Math.round((value / Math.max(total, localGoal)) * 100), 100),
+          tooltip: showTooltips ? (localGoal === 0 ? "-" : `${Math.round((value / localGoal) * 100)}% ${labels[idx].toLowerCase()}`) : null,
+        })),
+      );
+
+      if (total > goal) {
+        setX100(Math.round((goal / total) * 100));
+      } else {
+        setX100(null);
+      }
+    } else {
+      setBars([]);
+      setTotal(0);
+      setX100(null);
+    }
+  }, [values, goal, labels, showTooltips]);
```

Comment:
Pas besoin de useState et de useEffect, c'est seulement du derived state. Si les valeurs sont modifiées par le composant parent (le nb de places), il sera re-rendu, donc ce composant aussi, et donc les valeurs mises à jour automatiquement.
```suggestion
  const total = values.reduce((value, originalValue) => value + originalValue, 0);
  const localGoal = goal === 0 ? total : goal;
  const x100 = localGoal === 0 ? null : Math.round((goal / total) * 100);
  const colors = ["#1D4ED8", "#E5E7EB"];
  const bars = values.map((value, idx) => ({
    color: colors[idx % colors.length],
    label: labels[idx],
    value,
    percent: localGoal === 0 ? "-" : `${Math.round((value / localGoal) * 100)}%`,
    width: Math.min(Math.round((value / Math.max(total, localGoal)) * 100), 100),
    tooltip: showTooltips ? (localGoal === 0 ? "-" : `${Math.round((value / localGoal) * 100)}% ${labels[idx].toLowerCase()}`) : null,
  }));
```